src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1388,7 +1388,7 @@ internal static int FromHex(char digit)
                     : ((int)digit - (int)'a'))
                     + 10);
             }
-            throw new ArgumentException(nameof(digit));
+            throw new ArgumentException(SR.Argument_NotHexDigit, nameof(digit));
```

Comment:
We tend to be very careful with Uri as even small innocent changes in behavior have caused problems in the past.

With respect to the derived exception types, I had not considered that. I agree we could dispense with adding resource strings and instead use a derived exception type like ArgumentOutOfRangeException.